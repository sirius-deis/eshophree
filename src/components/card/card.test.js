import { render, screen } from "@testing-library/react";
import { BrowserRouter as Router } from "react-router-dom";
import Card from "./card";


describe("Card component", () => {
  const defaultValues = {
    _id: "21fd434hd3",
    name: "Hero",
    images: ["test.jpg"],
    category: { _id: "5635geet3gfdgd", name: "Category" },
    price: 100,
    discount: 10,
    total: 10,
    sold: 5,
    endIn: 100000,
  };
  it("should match snapshot", () => {
    const { container } = render(<Router><Card {...defaultValues} /></Router>);
    expect(container).toMatchSnapshot();
  });

  it("should render a name", () => {
    render(<Router><Card {...defaultValues} /></Router>);
    expect(screen.getByText("Hero")).toBeInTheDocument();
    expect(screen.getByText('Category')).toBeInTheDocument();
  });

  it("should render an image", () => {
    render(<Router><Card {...defaultValues} /></Router>);
    expect(screen.getByRole("img")).toHaveAttribute("src", defaultValues.images[0]);
  });

  it("should render a product link", () => {
    render(<Router><Card {...defaultValues} /></Router>);
    expect(screen.getByText("Hero")).toHaveAttribute("href", `/product/${defaultValues._id}`);
  });
  it("should render a category link", () => {
    render(<Router><Card {...defaultValues} /></Router>);
    expect(screen.getByText("Category")).toHaveAttribute("href", `/category/${defaultValues.category._id}`);
  });
  it('calculates and displays discounted price', () => {
    render(<Router><Card {...defaultValues} /></Router>);
    expect(screen.getByText('$100.00')).toBeInTheDocument();
    expect(screen.getByText('$90.00')).toBeInTheDocument();
  });
});