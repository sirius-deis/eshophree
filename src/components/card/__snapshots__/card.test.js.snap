// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card component should match snapshot 1`] = `
<div>
  <div
    class="sc-bgqQcB lmryQB"
    style="flex-direction: row;"
  >
    <img
      alt="product"
      src="test.jpg"
    />
    <div
      class="sc-gTRrQi bnzzvf"
    >
      <a
        href="/category/5635geet3gfdgd"
      >
        Category
      </a>
      <a
        href="/product/21fd434hd3"
        style="font-size: 2.5rem;"
      >
        Hero
      </a>
      <p
        class="price"
      >
        $
        100.00
        <sup>
          <del>
            $
            90.00
          </del>
        </sup>
      </p>
    </div>
  </div>
</div>
`;
